chain-signatures/node/src/indexer_eth/mod.rs

```diff
@@ -0,0 +1,1514 @@
+                        );
+                    }
+                }
+            } else {
+                // If we don't have the type tracked, just remove it (fallback behavior for backward compatibility)
+                tracing::debug!(
+                    ?sign_id,
+                    "sign request type not found, removing from backlog"
+                );
+                backlog.remove(Chain::Ethereum, &sign_id).await;
+            }
+        }
+    }
+}
+
+fn sign_id_from_signature_responded_log(log: &Log) -> Option<SignId> {
+    if log
+        .topic0()
+        .is_none_or(|topic| *topic != SignatureResponded::SIGNATURE_HASH)
+    {
+        return None;
+    }
+
+    let request_topic = log.topics().get(1)?;
+    let request_id: [u8; 32] = (*request_topic).into();
+    Some(SignId { request_id })
+}
+
+fn send_indexed_requests_to_sign_queue(
+    requests: Vec<IndexedSignRequest>,
+    sign_tx: mpsc::Sender<Sign>,
+    node_near_account_id: AccountId,
+) {
+    for request in requests {
+        let sign_tx = sign_tx.clone();
+        let node_near_account_id = node_near_account_id.clone();
+        tokio::spawn(async move {
+            match sign_tx.send(Sign::Request(request)).await {
+                Ok(_) => {
+                    crate::metrics::NUM_SIGN_REQUESTS
+                        .with_label_values(&[
+                            Chain::Ethereum.as_str(),
+                            node_near_account_id.as_str(),
+                        ])
+                        .inc();
+                }
+                Err(err) => {
+                    tracing::error!(?err, "Failed to send ETH sign request into queue");
+                }
+            }
+        });
+    }
+}
+
+#[derive(Debug)]
+struct SignatureRequestedEvent {
+    requester: Address,
+    payload_hash: [u8; 32],
+    path: String,
+    key_version: u32,
+    deposit: U256,
+    chain_id: U256,
+    algo: String,
+    dest: String,
+    params: String,
+}
+
+impl SignatureRequestedEvent {
+    fn encode_abi(&self) -> Vec<u8> {
+        let signature_requested_event_encoding = SignatureRequestedEncoding {
+            sender: self.requester,
+            payload: self.payload_hash.into(),
+            path: self.path.clone(),
+            keyVersion: self.key_version,
+            chainId: self.chain_id,
+            algo: self.algo.clone(),
+            dest: self.dest.clone(),
+            params: self.params.clone(),
+        };
+        signature_requested_event_encoding.encode_data()
+    }
+
+    pub fn generate_request_id(&self) -> [u8; 32] {
+        let abi_encoded = self.encode_abi();
+        alloy::primitives::keccak256(abi_encoded).into()
+    }
+}
+
+#[derive(Clone)]
+pub enum EthereumClient {
+    Helios(indexer_eth_helios::HeliosEthereumClient),
+    DirectRpc(indexer_eth_direct_rpc::RpcEthereumClient),
+}
+
+#[async_trait]
+impl EthereumClientTrait for EthereumClient {
```

Comment:
We don't need `EthereumClientTrait` at all since `EthereumClient` is an enum. Just implement these functions directly on `EthereumClient`